android-plugin/src/main/java/io/ecocode/java/checks/social/GoogleTrackerRule.java

```diff
@@ -0,0 +1,153 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package io.ecocode.java.checks.social;
+
+import io.ecocode.java.checks.helpers.TreeHelper;
+import org.sonar.api.issue.Issue;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Check the presence of the import com.google.android.gms.analytics.Tracker
+ * Importing the com.google.android.gms.analytics.Tracker class means that
+ * the app sends hits to Google Analytics. It is not necessarily sensitive
+ * information, but it is a first step towards Google Ads and hence this
+ * practice should be discouraged at early stage.
+ */
+
+@Rule(key = "SPRI002", name = "ecocodeGoogleTracker")
+public class GoogleTrackerRule extends BaseTreeVisitor implements JavaFileScanner {
+
+    private static final String ERROR_MESSAGE_GGL_TRACKER = "Using com.google.android.gms.analytics.* is a potential threat for privacy.";
+    private static final String ERROR_MESSAGE_FIREBASE_TRACKER = "Using com.google.firebase.analytics.* is a potential threat for privacy.";
+
+    @Override
+    public void scanFile(JavaFileScannerContext context) {
+        CompilationUnitTree cut = context.getTree();
+
+        GoogleTrackerImports googleTrackerImport = new GoogleTrackerImports();
+
+        for (ImportClauseTree importClauseTree : cut.imports()) {
+            ImportTree importTree = null;
+
+            if (importClauseTree.is(Tree.Kind.IMPORT)) {
+                importTree = (ImportTree) importClauseTree;
+            }
+
+            if (importTree == null) {
+                // discard empty statements, which can be part of imports
+                continue;
+            }
+
+            googleTrackerImport.collectTrackerImport(importTree);
+        }
+
+        handleResult(context, googleTrackerImport);
+        scan(cut);
+    }
+
+    private void handleResult(JavaFileScannerContext context, GoogleTrackerImports googleTrackerImport) {
+
+        //TODO FIX if firebase import is after google import in the same file
+        if (googleTrackerImport.hasTrackerImports()) {
+            googleTrackerImport.allAnalyticsImport = new ArrayList<>();
+            googleTrackerImport.allAnalyticsImport.addAll(googleTrackerImport.getGoogleTrackerImports());
+
+            googleTrackerImport.allAnalyticsImport.addAll(googleTrackerImport.getFirebaseTrackerImports());
```

Comment:
Can we have a getter in GoogleTrackerImports  that directly returns the combined list?